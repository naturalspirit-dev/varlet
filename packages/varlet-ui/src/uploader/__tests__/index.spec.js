const Uploader = require('../../../cjs/uploader').default
const { render } = require('@testing-library/vue')

test('test uploader', async () => {
  const wrapper = render(Uploader)
  console.log(wrapper)
})
