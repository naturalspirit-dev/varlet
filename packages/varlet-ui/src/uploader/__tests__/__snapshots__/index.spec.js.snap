// Vitest Snapshot v1

exports[`test uploader disabled 1`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__action var-elevation--2\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test uploader disabled 2`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__file var-elevation--2\\">
      <div class=\\"var-uploader__file-name\\">cat.png</div>
      <div class=\\"var-uploader__file-close\\"><i class=\\"var-icon var-icon--set var-icon-delete var-uploader__file-close-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
      <!--v-if-->
      <div class=\\"var-uploader__file-indicator\\"></div>
    </div>
    <div class=\\"var-uploader__action var-elevation--2 var-uploader--disabled\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\" disabled=\\"\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test uploader hide file list 1`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__action var-elevation--2\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test uploader length over maxlength in multiple mode 1`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__action var-elevation--2\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\"></div>
      <div class=\\"var-form-details__length\\">0 / 1</div>
    </div>
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test uploader validation 1`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__action var-elevation--2\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\">您至少上传一个</div>
      <div class=\\"var-form-details__length\\"></div>
    </div>
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test uploader validation 2`] = `
"<div class=\\"var-uploader var--box\\">
  <div class=\\"var-uploader__file-list\\">
    <div class=\\"var-uploader__file var-elevation--2\\">
      <div class=\\"var-uploader__file-name\\">cat.png</div>
      <div class=\\"var-uploader__file-close\\"><i class=\\"var-icon var-icon--set var-icon-delete var-uploader__file-close-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
      <!--v-if-->
      <div class=\\"var-uploader__file-indicator\\"></div>
    </div>
    <div class=\\"var-uploader__action var-elevation--2\\"><input class=\\"var-uploader__action-input\\" type=\\"file\\" accept=\\"image/*\\"><i class=\\"var-icon var-icon--set var-icon-plus var-uploader__action-icon\\" style=\\"transition: transform 0ms;\\" var-uploader-cover=\\"\\"></i></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;
